### ISSUE TITLE
Nextion Driver won't shut down

### ISSUE THREAD

**Issue by umj** [on7lds/NextionDriver#13]
I recently performed an MMDVM Firmware update , I closed the nextion driver  it indicated it was closed but obviously remained open, preventing the firmware update to load, corrupting the SD card, I re-flashed the sd card with a new pi-star image and ran the firmware update without an issue, afterwards I reloaded the nextion driver for a 2.5 display.

Now that I I know this is an issue with the driver, Ill prepare a new card untill the issue with the driver is corrected.

Rick N9UMJ

**Comment by on7lds** [comment]
I can not repeat your problem.
I doubt the SD card gets corrupted by closing NextionDriver.
What I already experienced is that when you have a Pi running for quite some time with a 'normal' distribution, you could have a corrupted card without knowing at first. But when rebooting it shows.
This is not the case with distributions for embedded systems (like PiStar) who keep the card in read-only the whole time (and only for specific reasons like updates it is set in read/write).

**Comment by umj** [comment]
not sure! I ve had it do the same thing on 2 different hotspots and 2
different cards, Both were on duplex hotspots,  One of them built by
Winteres Bi7jta and the other by Florian DF2ET.
I did mention it to Winters and he said others have experienced the same
problem and pointed me to the Nextion Driver still operational when trying
to make a firmware upgrade. When I formatted and created a new image and
did the upgrade first , then load Nextion Drivers, This worked on both
hotspots.

This is the error I was receiving when trying to make the upgrade, I tried
sudo service nextiondriver stop  before starting the upgrade as suggested
in other discussions.


[image: image.png]

Thats when it was suggested that the nextion driver was not shutting down.
and posted the issue.

"OK,
I can ensure it the Nextion serial data inject issue ,have other ham
feedback and I got issue some time.

I plan to submit a issue here , you can also do this .
https://github.com/on7lds/NextionDriver/issues?q=is%3Aissue+is%3Aclosed
*  "*







All is good now,

Thank You for looking for the problem and your reply

73 Rick n9umj

**Comment by bi7jta** [comment]
Hi ON7LDS,
I meet this issue ever, not one time. the UDP inject data to ttyAMA0 will interfere  with the firmware upgrade process, or make the Modem can not connect, I try to run "sudo service nextiondriver stop" ,but not success, seem the nextiondriver service not real kill , just when I rename "/usr/local/bin/NextionDriver" to "..._bak", and reboot , it will recover normal again.

Then I reinstall NextionDriver again, this issue gone. I do not know what cause this issue, hope you can fix it. 

[Transparent Data]
Enable=1
SendFrameType=1
RemoteAddress=127.0.0.1
RemotePort=40094
LocalPort=40095

DE BI7JTA

**Comment by umj** [comment]
Hello all

It was nice to know that there was an issue, I was able to save a backup on
my zumspot.

By flashing a new card , doing the firmware upgrade and then installing the
backup , made the upgrade smoother and less frustrating than the first 2
hotspot upgrades, I appreciate all that both if you do for the hobby.

Thank you both for help resolving this issue.

Rick n9umj

**Comment by on7lds** [comment]
Today I did several PiStar installs, then NextionDriver install, then PiStar updates and upgrades ...
Not once a problem.

Which Pi, wich PiStar version do you use ? Pls also send me PiStar config backup

**Comment by umj** [comment]
Im using a pi 3 b+ and pi-star v 4.1.3 , recently upgrade from 4.1.2

attached is the backup,
It's under my wife's call on the hotspot here at home, The other hotspot I
use at work under my callsign, The are identical on displays 2.5" pi and
pistar versions,
The only difference is the hotspots One was made by Bi7jta and the other by
DF2ET and DB9MAT. Both are duplex hotspots, SD cards are the same as well
Sandsk 32 GB

Rick n9umj
